bundle/edu.gemini.util.security/src/main/java/org/h2/engine/Engine.java

```diff
@@ -67,11 +72,15 @@ private Session openSession(ConnectionInfo ci, boolean ifExists, String cipher)
                 DATABASES.put(name, database);
             }
         }
+
+        logger.info("Attempt to hold a lock on " + database);
         synchronized (database) {
+            logger.info("Lock on " + database + " acquired, db opened: " + opened);
             if (opened) {
                 // start the thread when already synchronizing on the database
                 // otherwise a deadlock can occur when the writer thread
                 // opens a new database (as in recovery testing)
+                logger.info("Waiting for database to open: " + name);
                 database.opened();
             }
             if (database.isClosing()) {
```

Comment:
I'd like to see a message here as well.  One theory is that this method is returning `null` because of this.